include/cpu/cpus.h

```diff
@@ -30,25 +29,26 @@ void cpu_stop_current(void);
 void cpu_synchronize_all_post_reset(void);
 void cpu_synchronize_all_post_init(void);
 
+typedef struct TimersStateScale {
+    int32_t clock_scale_enable;
+    int32_t clock_scale_factor;
+} TimersStateScale;
+
 typedef struct TimersState {
     int64_t cpu_ticks_prev;
     int64_t cpu_ticks_offset;
     int64_t cpu_clock_offset;
     int32_t cpu_ticks_enabled;
     int64_t dummy;
 
-    /* slow down vm clock by a factor x */
-    int32_t clock_scale_enable;
-    int32_t clock_scale;
+    /* slow down vm clock by a factor x. This is shared with QEMU over the libs2e interface */
+    TimersStateScale *cpu_clock_scale;
```

Comment:
Don't store pointers here. IIRC, TimersState is a per-state structure.

Reply:
Done. I think

Reply:
Done, I think?